client/init.c

```diff
@@ -216,6 +226,32 @@ TDNFRefreshSack(
         {
             continue;
         }
+        nCount++;
+    }
+
+    /* nCount may be 0 if --disablerepo=* is used */
+    if (nCount > 0)
```

Comment:
Not sure what I am missing - the block starting in line 220 is a loop for counting the repos. Only after counting all (valid) repos we know how much memory to allocate and whether to sort, or skip that step when there is no valid repo.

Reply:
I thought instead of allocating for full array after taking nCount, we can allocate and reallocate where we do nCount now. Never mind, the existing version is better, memory allocations are costly operations.